src/main/java/com/google/devtools/build/lib/profiler/Profiler.java

```diff
@@ -138,27 +138,47 @@ public ProfilerTask getType() {
     final long startTimeNanos;
     final int id;
     final ProfilerTask type;
+    final MnemonicData mnemonic;
     final String description;
 
     long duration;
 
-    TaskData(int id, long startTimeNanos, ProfilerTask eventType, String description) {
+    TaskData(int id, long startTimeNanos, ProfilerTask eventType, MnemonicData mnemonic, String description) {
       this.id = id;
       this.threadId = Thread.currentThread().getId();
       this.startTimeNanos = startTimeNanos;
       this.type = eventType;
+      this.mnemonic = mnemonic;
       this.description = Preconditions.checkNotNull(description);
     }
 
+    TaskData(int id, long startTimeNanos, ProfilerTask eventType, String description) {
+      this(id, startTimeNanos, eventType, MnemonicData.getEmptyMnemonic(), description);
+    }
+
     TaskData(long threadId, long startTimeNanos, long duration, String description) {
       this.id = -1;
       this.type = ProfilerTask.UNKNOWN;
+      this.mnemonic = MnemonicData.getEmptyMnemonic();
       this.threadId = threadId;
       this.startTimeNanos = startTimeNanos;
       this.duration = duration;
       this.description = description;
     }
 
+    /**
+     * Get the value of the datum's "cat" field for JSON output.
+     *
+     * @return the category value if one can be created, otherwise null
+     */
+    public String getJsonCategory() {
```

Comment:
I don't think that function carries its own weight.

Reply:
While it doesn't do much, I left this in after reverting the use of the "cat" field since I think it's a bit clearer to use this than to access `data.type.description` directly. I do admit that it's a bit out of place since more clarity could be brought with a larger refactoring, which would be a separate issue from this adding of mnemonics, so I can revert this as well if you still think it's apt to do so.